iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -288,7 +309,11 @@ template <typename T>
 template <typename... Targs>
 inline void optional<T>::construct_value(Targs&&... args) noexcept
 {
-    new (static_cast<T*>(static_cast<void*>(&m_data))) T(std::forward<Targs>(args)...);
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8 : The optional has the type T defined
+    // during compile time and the type is unchangeable during the lifetime of the object.
+    // All accesses to the underlying data is done via the same static type and therefore the
+    // casts are always valid
+    new (static_cast<T*>(static_cast<void*>(m_data))) T(std::forward<Targs>(args)...);
```

Comment:
`&` not needed. Also the cast s are not needed here, `new (&m_data) ... `. Note that the pointer is already aligned for T by design. Maybe this makes the comment unnecessary.